br/pkg/lightning/restore/get_pre_info.go

```diff
@@ -665,17 +669,32 @@ outloop:
 		switch errors.Cause(err) {
 		case nil:
 			if !initializedColumns {
+				ignoreColsMap := igCols.ColumnsMap()
 				if len(columnPermutation) == 0 {
 					columnPermutation, err = createColumnPermutation(
 						columnNames,
-						igCols.ColumnsMap(),
+						ignoreColsMap,
 						tableInfo,
 						log.FromContext(ctx))
 					if err != nil {
 						return 0.0, false, errors.Trace(err)
 					}
 				}
+				if len(ignoreColsMap) > 0 || len(sampleFile.ExtendData.Columns) > 0 {
+					_, extendCols, extendVals = filterColumns(columnNames, sampleFile.ExtendData, ignoreColsMap, tableInfo)
```

Comment:
in current code of `filterColumns`, `ignoreColsMap` will not affect the returned `extendCols, extendVals`

Reply:
addressed